### ISSUE TITLE
Guide - Routes

### ISSUE THREAD

**Issue by crisward** [kemalcr/kemalcr.com#14]
There is a line in the guide which says

>Routes are matched in the order they are defined. The first route that matches the request is invoked.

I'm not sure this is correct. Radix won't allow two identical routes, and those with overlap, the more specific tends to win... I think? Perhaps @luislavena can confirm.

**Comment by sdogruyol** [comment]
Yeah, It doesn't allow two identical routes. We need to document it in a more explicit way

**Comment by luislavena** [comment]
Hello, apologies for late reply...

> Radix won't allow two identical routes, and those with overlap, the more specific tends to win... I think?

That's actually not accurate. [Radix](https://github.com/luislavena/radix) will not allow two equal routes but will allow adding a _payload_ to a shorter one, example:

```crystal
require "radix"

tree = Radix::Tree(Symbol).new

# this works
tree.add "/foo/bar/baz", :baz
tree.add "/foo/bar", :bar

# not found (no payload at `/foo`)
result = tree.find "/foo"
pp result.found? # => false

result = tree.find "/foo/bar/baz"
pp result.found?  # => true
pp result.payload # => :baz

result = tree.find "/foo/bar"
pp result.found?  # => true
pp result.payload # => :bar
```

Hope that helps.

Cheers.
